matRad_calcDoseDirect.m

```diff
@@ -52,16 +52,20 @@
 end
 
 % dose calculation
-if strcmp(pln.radiationMode,'photons')
-    dij = matRad_calcPhotonDose(ct,stf,pln,cst,calcDoseDirect);
-    %dij = matRad_calcPhotonDoseVmc(ct,stf,pln,cst,5000,4,calcDoseDirect);
-elseif strcmp(pln.radiationMode,'protons') || strcmp(pln.radiationMode,'carbon')
-    dij = matRad_calcParticleDose(ct,stf,pln,cst,calcDoseDirect);
+dij = cell(length(stf),1);
```

Comment:
I dont like this because this will be confusing when we merge this with the code from the robust optimization branch. We would end up with dij{beamNum}.physicalDose{ScenNum}. Additionally, the current code will not work for a recalculation using a variable RBE as in the case for carbon ions. For N beams you would have to run the direct dose calculation function N+1 times. +1 to consider mixed field effects. Adding up RBEweighted beam doses will not work.  I would suggest, that we make the beam dose calculation optional for direct dose calc.  If someone needs to look at beam doses after a direct dose calculation then a full dij should be calculated - in that case simply set calcDoseDirect to false. Overall the code might be a bit slower, but better structured. This would clear up the code because the rest of the file would basically stay the same. You only need to differentiate between calcDoseDirect == true or false.   If false then simply call the existing function matRad_calcCubes and if calcDoseDirect == true then run the existing code.  I am going to create a pull request.